src/core/mac/mac_frame.cpp

```diff
@@ -450,6 +488,26 @@ void Frame::SetSrcAddr(const Address &aAddress)
     }
 }
 
+otError Frame::GetSecurityControlField(uint8_t &aSecurityControlField) const
+{
+    otError error = OT_ERROR_NONE;
+    uint8_t index = FindSecurityHeaderIndex();
+    VerifyOrExit(index != kInvalidIndex, error = OT_ERROR_PARSE);
+
+    aSecurityControlField = GetPsdu()[index];
+
+exit:
+    return error;
+}
+
+void Frame::SetSecurityControlField(uint8_t aSecurityControlField)
+{
+    uint8_t index = FindSecurityHeaderIndex();
+    OT_ASSERT(index != kInvalidIndex);
```

Comment:
why do we return error in `GetSecurityControlField` but assert here?

Reply:
Then I prefer to also add error return here. Thoughts?

Reply:
I think set failure is logic error, which should use assert. Get failure maybe caused by malformed input. @simonlingoogle